exporters/autoexport/metrics.go

```diff
@@ -94,4 +98,43 @@ func init() {
 	RegisterMetricReader("none", func(ctx context.Context) (metric.Reader, error) {
 		return newNoopMetricReader(), nil
 	})
+	RegisterMetricReader("prometheus", func(ctx context.Context) (metric.Reader, error) {
+		host := os.Getenv("OTEL_EXPORTER_PROMETHEUS_HOST")
+		if host == "" {
+			host = "localhost"
+		}
+
+		port := os.Getenv("OTEL_EXPORTER_PROMETHEUS_PORT")
+		if port == "" {
+			port = "9464"
+		}
+
+		go serve(ctx, &http.Server{
+			Addr:         host + ":" + port,
+			ReadTimeout:  5 * time.Second,
+			WriteTimeout: 10 * time.Second,
+			IdleTimeout:  120 * time.Second,
+		})
+
+		return prometheus.New()
+	})
+}
+
+func serve(ctx context.Context, server *http.Server) {
```

Comment:
Have you tested this?  I don't think you are serving prometheus metrics on the server:
https://github.com/open-telemetry/opentelemetry-go/blob/d18277e3c63f64cb1d94608d561786a632187d0f/example/prometheus/main.go#L111

Reply:
I deleted the key part while refactoring, and my tests didn't save me :disappointed: .

Please take another look.

Reply:
I need to make the test more robust (wait for the server to be ready).

Reply:
Looks better, thanks!